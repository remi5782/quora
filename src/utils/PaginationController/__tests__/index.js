import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import PaginationControl from '../index';
import renderer from 'react-test-renderer';

test('Checking Pagination Comp  with different inputs', () => {
    let checkingProps = {gotoPage:()=>{}, nbPages:12}
    let component = renderer.create(
        <BrowserRouter>
        <PaginationControl {...checkingProps}/>
        </BrowserRouter>
        
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    checkingProps = {gotoPage:()=>{}, nbPages:null}
    component = renderer.create(
        <BrowserRouter>
        <PaginationControl {...checkingProps}/>
        </BrowserRouter>
        
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    
});