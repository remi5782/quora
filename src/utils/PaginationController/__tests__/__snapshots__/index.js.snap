// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card  props with different values 1`] = `
<div
  className="makeStyles-root-1"
>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    Page: 
    1
  </p>
  <nav
    aria-label="pagination navigation"
    className="MuiPagination-root"
  >
    <ul
      className="MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-current="true"
          aria-label="page 1"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary Mui-selected"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          1
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 2"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          2
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 3"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          3
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 4"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          4
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 5"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          5
        </button>
      </li>
      <li>
        <div
          className="MuiPaginationItem-root MuiPaginationItem-ellipsis"
        >
          â€¦
        </div>
      </li>
      <li>
        <button
          aria-label="Go to page 11"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          11
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Card  props with different values 2`] = `
<div
  className="makeStyles-root-1"
>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    Page: 
    1
  </p>
  <nav
    aria-label="pagination navigation"
    className="MuiPagination-root"
  >
    <ul
      className="MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-current="true"
          aria-label="page 1"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary Mui-selected"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          1
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Checking Pagination Comp  with different inputs 1`] = `
<div
  className="makeStyles-root-1"
>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    Page: 
    1
  </p>
  <nav
    aria-label="pagination navigation"
    className="MuiPagination-root"
  >
    <ul
      className="MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-current="true"
          aria-label="page 1"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary Mui-selected"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          1
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 2"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          2
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 3"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          3
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 4"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          4
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 5"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          5
        </button>
      </li>
      <li>
        <div
          className="MuiPaginationItem-root MuiPaginationItem-ellipsis"
        >
          â€¦
        </div>
      </li>
      <li>
        <button
          aria-label="Go to page 11"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          11
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Checking Pagination Comp  with different inputs 2`] = `
<div
  className="makeStyles-root-1"
>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    Page: 
    1
  </p>
  <nav
    aria-label="pagination navigation"
    className="MuiPagination-root"
  >
    <ul
      className="MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-current="true"
          aria-label="page 1"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary Mui-selected"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          1
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-outlinedSecondary Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
